### ISSUE TITLE
Modified %{_smp_mflags} breaks find-debuginfo.sh

### ISSUE THREAD

**Issue by xsuchy** [rpm-software-management/rpm#630]
On my system I have user defined:
```
rpm --eval '%{?_smp_mflags}'
-l4
```
which for Makefile has semantics:
```
       -l [load], --load-average[=load]
            Specifies  that  no  new jobs (commands) should be started if there are others jobs running and the load average is at least load (a floating-point
            number).  With no argument, removes a previous load limit.
```
But find-debuginfo.sh does not recognize it and interpret it differently (as BUILDDIR)

`rpmbuild` will then fail with:

```
Processing files: enum-debugsource-1.1-15.fc29.x86_64
error: Could not open %files file /home/msuchy/rpmbuild/BUILD/enum-1.1/debugsourcefiles.list: No such file or directory
```

Reproducer is easy. Put in your `~/.rpmmacros` this definition:

```
%_smp_mflags -l4
```

**Comment by xsuchy** [comment]
Note that `-l` is e.g., used at this example https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/rpm_packaging_guide/advanced-topics#custom-macros

**Comment by pmatilai** [comment]
Oh ugh. Whoever wrote that piece of advise about %_smp_mflags ... <cencored>

**Comment by pmatilai** [comment]
Anyway, find-debuginfo *could* be smarter about it's arguments, something like this ought to fix it:
```
--- a/scripts/find-debuginfo.sh
+++ b/scripts/find-debuginfo.sh
@@ -190,9 +190,11 @@ while [ $# -gt 0 ]; do
     srcout=$2
     shift
     ;;
+  -*)
+    echo >&2 "warning: unknown option $1"
+    ;;
   *)
     BUILDDIR=$1
-    shift
     break
     ;;
   esac
```

**Comment by pmatilai** [comment]
Actually... while adding some error handling to find-debuginfo.sh argument parsing wouldn't hurt, a more resilient fix is to eliminate use of %_smp_mflags in that context. Since commit e811c7ec0b4d2685b63b61803e3952466b1a4ac6 we can trivially do just that.

**Comment by xsuchy** [comment]
Well `--load-average` *is* valid smp macro.

I just *briefly* checked code of rpmbuild and it seems to me that `-l` is not actively used. So what about parsing `-l` and `--load-average` similarly as `-j` and `--jobs` and current `-l` rename to `--list` or `--filelist`? 

Other option I see is to use:
```
  -l)
    lists[$nout]="${lists[$nout]} $2"
    shift
    ;;
  -l*)
    n_jobs=${1#-j}
    ;;

```
It is a bit weird as it would overload `-l` or we can put there just no-op, so `-l*` will be ignored. It is still better than failing.

**Comment by pmatilai** [comment]
Fixed in commit f7282c3, thanks for the report!

**Comment by xsuchy** [comment]
Wow, that was quick. Thank you, Panu.
